test/common/http/http2/codec_impl_test.cc

```diff
@@ -45,6 +45,28 @@ namespace CommonUtility = ::Envoy::Http2::Utility;
 
 class Http2CodecImplTestFixture {
 public:
+  // The Http::Connection::dispatch method does not throw (any more). However unit tests in this
```

Comment:
What is the end goal of this? My read on this is that after exceptions are removed this will have to stay, because otherwise we will have to plumb the error statuses through write.